cli/cobra.go

```diff
@@ -161,6 +161,20 @@ func (tcmd *TopLevelCommand) Initialize(ops ...command.InitializeOpt) error {
 	return tcmd.dockerCli.Initialize(tcmd.opts, ops...)
 }
 
+// ContextType return the current context type, "" for default
```

Comment:
question: just thinking out of loud, but what about returning "engine", "docker" or "moby" by default?

Reply:
This part will have to change when rebasing this PR on top of https://github.com/docker/cli/pull/2895, adding a cleaner way to get context type.